posts/2024-08-27-24.0.0.9-beta.adoc

```diff
@@ -0,0 +1,212 @@
+seo-description: In this release, the MicroProfile Telemetry 2.0 beta feature helps improve the observability of your Java applications by using OpenTelemetry to collect and export logs, metrics, and traces in a standardized way. A minor update for the WebSocket 2.2 beta feature is also included.
+blog_description: In this release, the MicroProfile Telemetry 2.0 beta feature helps improve the observability of your Java applications by using OpenTelemetry to collect and export logs, metrics, and traces in a standardized way. A minor update for the WebSocket 2.2 beta feature is also included.
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+---
+= Send logs and metrics to OpenTelemetry in 24.0.0.9-beta
+David Mueller <https://github.com/dmuelle>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//
+
+In this release, the MicroProfile Telemetry 2.0 beta feature helps improve the observability of your Java applications by using OpenTelemetry to collect and export logs, metrics, and traces in a standardized way. A minor update for the WebSocket 2.2 beta feature is also included.
+
+The link:{url-about}[Open Liberty] 24.0.0.9-beta includes the following beta features (along with link:{url-prefix}/docs/latest/reference/feature/feature-overview.html[all GA features]):
+
+* <<metrics, Send Liberty metrics to OpenTelemetry>>
+* <<logs, Send logs to OpenTelemetry>>
+* <<websocket, Jakarta EE11 WebSocket 2.2>>
+
+
+See also link:{url-prefix}/blog/?search=beta&key=tag[previous Open Liberty beta blog posts].
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/29456
+// Contact/Reviewer: Channyboy
+// // // // // // // //
+[#metrics]
+== Send Liberty metrics to OpenTelemetry
+
+MicroProfile Telemetry 2.0 provides you with the latest Open Telemetry technology. In addition to link:{url-prefix}/docs/latest/microprofile-telemetry.html[distributed tracing], the feature now allows OpenTelemetry to collect and export metrics and logs.
+
+When the MicroProfile Telemetry 2.0 beta feature (`mpTelemetry-2.0`) is enabled, Open Liberty can now forward runtime component statistics that are captured by the link:{url-prefix}/docs/latest/reference/feature/monitor-1.0.html[Performance Monitoring 1.0] feature (`monitor-1.0`) to the MicroProfile Telemetry 2.0 runtime. This statistical data is registered as metrics in the telemetry runtime and can then be forwarded to any OpenTelemetry Protocol (OTLP) compatible metric consumer to meet your monitoring needs.
+
+The following runtime components are supported:
+
+* `ThreadPool`
+* `Sessions`
+* `RequestTiming`
+* `ConnectionPool`
+
+To collect and export metrics, enable OpenTelemetry by using the following system property or environment variable:
+
+* System property: `otel.sdk.disabled=false`
+* Environment variable: `OTEL_SDK_DISABLED=false`
+
```

Comment:
@dmuelle

For the metrics for Telemetry also worth adding the following (similiar to what was done for logging).

By default, all OpenTelemetry data is exported to [OTLP](https://opentelemetry.io/docs/languages/java/exporters/#otlp). You set a different exporter by specifying the following system property or environment variable:

System property: otel.metrics.exporter

Environment variable: OTEL_METRICS_EXPORTER

Additionally, you may wish to configure the [metric export interval](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#periodic-exporting-metricreader) configuration variable. The value is in milliseconds and the default is set to 60000 (i.e., 60 seconds).

System property: otel.metric.export.interval

Environment variable: OTEL_METRIC_EXPORT_INTERVAL

Reply:
updated